MdePkg/Test/Mock/Library/GoogleTest/Protocol/MockSmmAccess2.cpp

```diff
@@ -0,0 +1,27 @@
+/** @file MockSmmAccess2.cpp
+  Declare mock SMM Access2 Protocol.
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#include <GoogleTest/Protocol/MockSmmAccess2.h>
+
+MOCK_INTERFACE_DEFINITION (MockSmmAccess2Protocol);

```

Comment:
This mock can be done entirely in the header file.
Example: https://github.com/microsoft/mu_basecore/blob/dev/202405/MdePkg/Test/Mock/Include/GoogleTest/Protocol/MockFirmwareVolume2.h

Reply:
Please check the update.